### DISCUSSION TITLE
Event or a way to recognize damage

### DISCUSSION THREAD

**Discussion by FrahHS** [BG-Software-LLC/SSB-AcidIslands#10] [Ideas]
Hi, i need a Listener that have to listen the Damage event caused by acid water, i tryed for several hours to contribute the plugin AcidIsland but i didn'è figured out how to call a custom event inside a module.

I can't figure out how to call the Event in a correct way...

I just need an event in the API like "AcidDamageEvent"
or a way to identify the acid damage like an EntityDamageEvent.DamageCause custom implemented in the SS2 API
or a scoreboard tag
or something else...

**Comment by OmerBenGera** [comment]
EntityDamageEvent is your way to go.
Check if the DamageCause is CUSTOM and that the player is in water.

**Reply by FrahHS** [reply]
Thank you for your answer! 
I did exaxtly that, but there are 2 problems.
The thread of the damage continue to multiply the damage, and is a problem for my project.
And when the player jump in the water i don't know why but he get the damage (i wanna cancel the event), it is like a bug.

//Check if the entity is a player 
if(!(e.getEntity() instanceof Player)) 
           return; 
  
// Check if the damage is custom 
If(e.getCause() != EntityDamageEvent.DamageCause.CUSTOM) 
         return; 
  
Player player = (Player) e.getEntity(); 
  
// Check if the player is in water 
if(!PlayerUtils.isInWater(player)) 
        return;

I used the same function of SSBAcidIslands to check if the player is in the water

**Reply by FrahHS** [reply]
Any suggestion?

**Reply by OmerBenGera** [reply]
Nope. Add debug messages and find out if you actually cancel the event.

**Reply by FrahHS** [reply]
Is there no way to call an event using callEvent from a module?

**Reply by OmerBenGera** [reply]
You can do whatever you want with the Bukkit API as a module.

**Reply by FrahHS** [reply]
I forked SSB Acid Island to try to implement an Event, you can look at the [code here](https://github.com/FrahHS/SSB-AcidIslands)
i imported SSB-AcidIsland as dependency in another plugin and I can't listen for this event (I created the listener but is not working).

what am I doing wrong?

AcidDamageEvent acidDamageEvent = new AcidDamageEvent(player, lastDamage);
Bukkit.getPluginManager().callEvent(acidDamageEvent); 
if (!acidDamageEvent.isCancelled()) {
    player.damage(lastDamage);
    lastDamage *= plugin.getSettings().damageMultiplier;
}

P.S. i'm sure that everything else is working and i load the listener only if the module is found thro the SSB2 API

**Reply by FrahHS** [reply]
My error:
[MyPlugin] Plugin MyPlugin v1.0 has failed to register events for class com.frahhs.myplugin.listeners.AcidDamageListener because com/bgsoftware/ssbacidislands/event/AcidDamageEvent does not exist.

**Comment by OmerBenGera** [comment]
I already told you about the solution.
Making custom events will not work, you won't be able to access them as they are part of a module, which is not accessible to other plugins.
